verifiers/internal/gcb/provenance.go

```diff
@@ -357,7 +356,9 @@ func (self *Provenance) VerifySourceURI(expectedSourceURI string, builderID util
 		return fmt.Errorf("%w: no materials", serrors.ErrorInvalidDssePayload)
 	}
 	uri := materials[0].URI
-	if !strings.HasPrefix(expectedSourceURI, "https://") {
+
+	// It is possible that GCS builds at level 2 use GCS sources, prefixed by gs://.
+	if strings.HasPrefix(uri, "https://") && !strings.HasPrefix(expectedSourceURI, "https://") {
```

Comment:
nit: You might also leave a note as to why this is necessary in the first place. It's not necessarily obvious why "https://" might need to be prefixed at the beginning.

Reply:
Also, consider checking for any URI scheme rather than "https://" specifically. So you don't end up with strange URIs like "https://http://example.com/..." if the user enters "--source http://example.com"

```suggestion
	if strings.HasPrefix(uri, "https://") && !strings.HasPrefix(expectedSourceURI, "://") {
```